### DISCUSSION TITLE
How to set cell type?

### DISCUSSION THREAD

**Discussion by dorothythurston** [exceljs/exceljs#2486] [Q&A]
Given some data like:
```
const workbook = new Excel.Workbook();
const sheet = workbook.addWorksheet('My Sheet');
const rows = [
    [1, 'Bob', true],
    [2, 'Jake', false]
];

sheet.addRows(rows);
```

How can I make the sheet column types in the excel file end up being  Excel.ValueType.Number, Excel.ValueType.String, Excel.ValueType.ValueType.Boolean?

**Comment by dorothythurston** [comment]
Looking at the code it seems like in cell.js Exceljs is trying to infer the cell type by checking the typeof the js value. Is there anyway to set the column cell type ahead of time?

**Comment by SerMarkin** [comment]
[Value types](https://github.com/exceljs/exceljs#value-types)
You can set value in cell then cell in Excel will have special type.

**Comment by nikhilag** [comment]
Is there any update on this? It will be very useful for cases where number is being stored as string or when date timezone needs to be changed on server side.

**Comment by adnaanbheda** [comment]
@nikhilag Hey, did you work out a solution for this ?

**Comment by zivlevy** [comment]
you can change the cell's format by the following code (in this example I change all cells in col 1 (A) to text format:
`    const cell = worksheet.getColumn(1);
      cell.eachCell((c, rowNumber) => {
        c.numFmt = '@';
      });`

**Comment by xianchenxy** [comment]
> Given some data like:
> 
> ```
> const workbook = new Excel.Workbook();
> const sheet = workbook.addWorksheet('My Sheet');
> const rows = [
>     [1, 'Bob', true],
>     [2, 'Jake', false]
> ];
> 
> sheet.addRows(rows);
> ```
> 
> How can I make the sheet column types in the excel file end up being Excel.ValueType.Number, Excel.ValueType.String, Excel.ValueType.ValueType.Boolean?

@dorothythurston @nikhilag  @adnaanbheda  
set the cell type automatically by exceljs, like
```
worksheet.columns.forEach(col => {
    col.eachCell(cell => {
        cell.value = +cell.value // convert a string value to a number value
    })
})
```

set the cell type by ourselves, using `numFmt`，like
```
    cell.numFmt = '0.00%' // convert `58` to `58.00%`
     // cell.numFmt = '#,##0' // convert '1234' to  `1,234`
```

or set total column, like
```
worksheet.getColumn(2).numFmt = '#,##0'
```
